[WIP ] #151 Edit events

abdellani opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

#### Issue addressed
<!-- include `fixes #<issue-number>` -->
<!-- the relevant issue in https://github.com/AgileVentures/WebsiteOne-FE/issues  -->
https://github.com/AgileVentures/WebsiteOne-FE/issues/151


#### Screenshots (if appropriate):
<!-- please include screenshots of any changes to the UI for quick review  -->
<!-- please show how things look on both desktop and mobile  -->

#### Testing
<!-- Remember you must see any new tests you created (or old ones you changed) -->
<!-- fail as well as pass in order to ensure they are working -->
<!-- Unsure how to test? - please see https://github.com/AgileVentures/WebsiteOne-FE/blob/develop/CONTRIBUTING.md#git-and-github-->

abdellani committed:
Adds edit Event button + route + form

abdellani commented:
@mattwr18 
At this level, users have access to the edit event form with data preloaded. Now, I wonder how to implement the change handlers for the form? is it better to do the changes at the component's state  or do I need to use actions/store  ?

abdellani committed:
Merge branch 'develop' of github.com:AgileVentures/WebsiteOne-FE into 151_edit_events

abdellani committed:
Implements change handling functions

abdellani committed:
Merge branch 'develop' of github.com:AgileVentures/WebsiteOne-FE into 151_edit_events

abdellani committed:
Implements partially handle submit.

Some corretions are needed:
-Not sure to which link the HTTP PUT request must be sent in updateEventAction.
-Is a reducer needed after the HTTP request (UPDATE_EVENT_INFO)?
-Cookies are not set up on this page.
-handleSubmit needs to be reviewed

abdellani committed:
Implements handle submit.

This commit was done during a pairing with @mattwr18

Video's link: https://youtu.be/oqKV0E2LbYE

Co-authored-by: mattwr18 <mattwr18@gmail.com>

abdellani committed:
Merge branch 'develop' of github.com:AgileVentures/WebsiteOne-FE into 151_edit_events

mattwr18 commented:
hey @abdellani there are 4 jest tests that are failing on the build, are they failing in the same way locally?

abdellani commented:
@mattwr18 
Is `/api/v1/events/${slug}` supposed to send data structured in the same way like in [cypress/fixtures/event.json](https://github.com/AgileVentures/WebsiteOne-FE/blob/develop/cypress/fixtures/event.json)  ? 
I'm asking because I faced a problem while writing cypress test. When I try to edit the event,  [src/actions/getEventInfoAction.js](https://github.com/AgileVentures/WebsiteOne-FE/blob/develop/src/actions/getEventInfoAction.js), The summary page doesn't show all the details, like the name for example.

Now, I'm using this [fixture](https://gist.github.com/abdellani/92d4967ebc7ae02888e339235521c8ab) for fetching event's info.

abdellani committed:
Merge branch 'develop' of github.com:AgileVentures/WebsiteOne-FE into 151_edit_events

abdellani committed:
Implements partially edit event cypress test. (Submitting changes is not handled yet)

abdellani committed:
Implement cypress test for edit event (A review is needed)

abdellani committed:
Passes EventInfo jest tests

abdellani committed:
Resolves EventSummary.test failures

mattwr18 commented on src/actions/updateEventAction.js:
```diff
@@ -0,0 +1,63 @@
+    }
+  })
+    .then(response => {
+      history.push(`/events/${response.data.event.slug}`)
```
We were talking about this is in the meeting @abdellani 
@joaopapereira has requested that we extract this concern from the action into either the reducer or the container...
so far we have other examples, i.e `Login.js`, where it is the concern of the container
Maybe, though we just want to open a ticket and do them all at the same time?

mattwr18 commented on src/containers/EditEventPage.js:
```diff
@@ -0,0 +1,149 @@
+    }
+    if (!this.props.projects.length) {
+      this.props.fetchActiveProjects()
+      this.props.fetchEventInfo(this.props.match.params.slug)
```
it seems a bit strange to have a `fetchEventInfo` nested inside an if block checking about the length of the projects array, doesn't it? I think I left it there in our pairing session?

mattwr18 commented on src/containers/EditEventPage.js:
```diff
@@ -0,0 +1,149 @@
+//       updateEvent
+//     }
+//   }
+// )(EditEventPage)
```
please remove the comments :smile_cat:

abdellani committed:
Implements jest tests

mattwr18 commented:
this is great @abdellani, I think we can get this merged soon... I'll pull down the latest changes and check it out.. did you see why the test coverage dropped?

abdellani committed:
Implements some changes according to comments.

The comments are :

https://github.com/AgileVentures/WebsiteOne-FE/pull/242/files/409401468c84e20255e5b9d4903d7dda9773729c#r287313707
https://github.com/AgileVentures/WebsiteOne-FE/pull/242/files/409401468c84e20255e5b9d4903d7dda9773729c#r287313989

mattwr18 commented on src/containers/EditEventPage.js:
```diff
@@ -101,7 +99,7 @@ export class EditEventPage extends Component {
         : <Fragment>
           <Header as='h1' textAlign='center'>
-          Creating a new Event
+            Creating a new Event
```
should this be `Editing Event`, or something similar @abdellani?

abdellani committed:
Implements 'UPDATE_EVENT_FAILURE' test for errorReducer

aonomike commented:
I think the test coverage dropped since there are no tests for `updateEventsAction.js`. Maybe an update to that would improve the coverage @abdellani @mattwr18  ?
